pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/ingress_test.go

```diff
@@ -0,0 +1,301 @@
+						{IP: "ip1", Ports: []v1.PortStatus{{Port: 80, Error: &errMsg}}},
+						{Hostname: "hostname1", Ports: []v1.PortStatus{{Protocol: "TCP", Port: 8080}}},
+						{IP: "ip2"},
+					},
+				},
+			},
+			expected: model.IngressStatus{
+				Ingress: []*model.LoadBalancerIngress{
+					{Ip: "ip1", Ports: []*model.PortStatus{{Port: 80, Error: "error"}}},
+					{Hostname: "hostname1", Ports: []*model.PortStatus{{Protocol: "TCP", Port: 8080}}},
+					{Ip: "ip2"},
+				},
+			},
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			assert.Equal(t, &tc.expected, extractIngressStatus(tc.input))
+		})
+	}
+}
+
+func TestExtractIngressRules(t *testing.T) {
+	pathType1 := netv1.PathTypeImplementationSpecific
+	pathType2 := netv1.PathTypeExact
+	tests := map[string]struct {
+		input    []netv1.IngressRule
+		expected []*model.IngressRule
+	}{
+		"empty": {input: []netv1.IngressRule{}, expected: []*model.IngressRule{}},
+		"multiple rules": {
+			input: []netv1.IngressRule{
+				{
+					Host: "host1",
+					IngressRuleValue: netv1.IngressRuleValue{
+						HTTP: &netv1.HTTPIngressRuleValue{
+							Paths: []netv1.HTTPIngressPath{
+								{Path: "path1-1", PathType: &pathType1},
+								{Path: "path1-2", PathType: &pathType1},
+							},
+						},
+					},
+				},
+				{
+					Host: "host2",
+					IngressRuleValue: netv1.IngressRuleValue{
+						HTTP: &netv1.HTTPIngressRuleValue{
+							Paths: []netv1.HTTPIngressPath{
+								{Path: "path2-1", PathType: &pathType2},
+								{Path: "path2-2", PathType: &pathType2},
+							},
+						},
+					},
+				},
+			},
+			expected: []*model.IngressRule{
+				{
+					Host: "host1",
+					HttpPaths: []*model.HTTPIngressPath{
+						{Path: "path1-1", PathType: "ImplementationSpecific", Backend: &model.IngressBackend{}},
+						{Path: "path1-2", PathType: "ImplementationSpecific", Backend: &model.IngressBackend{}},
+					},
+				},
+				{
+					Host: "host2",
+					HttpPaths: []*model.HTTPIngressPath{
+						{Path: "path2-1", PathType: "Exact", Backend: &model.IngressBackend{}},
+						{Path: "path2-2", PathType: "Exact", Backend: &model.IngressBackend{}},
+					},
+				},
+			},
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			assert.Equal(t, tc.expected, extractIngressRules(tc.input))
+		})
+	}
+}
+
+func TestExtractIngressBackend(t *testing.T) {
+	apiGroup := "apiGroup"
+	tests := map[string]struct {
+		input    netv1.IngressBackend
+		expected model.IngressBackend
+	}{
+		"empty": {input: netv1.IngressBackend{}, expected: model.IngressBackend{}},
+		"with resource": {
+			input: netv1.IngressBackend{
+				Resource: &v1.TypedLocalObjectReference{
+					APIGroup: &apiGroup,
+					Kind:     "kind",
+					Name:     "name",
+				},
+			},
+			expected: model.IngressBackend{
+				Resource: &model.TypedLocalObjectReference{
+					ApiGroup: "apiGroup",
+					Kind:     "kind",
+					Name:     "name",
+				},
+			},
+		},
+		"with service": {
```

Comment:
IIUC from the proto definition service name and port are mutually exclusive, do you think it'd make sense to split this in two to be more "realistic"?

Reply:
Actually, service and reference are mutually exclusive in `IngressBackend`. But a service and port can coexist in `IngressServiceBackend`. (I hope I understood the question correctly)

Reply:
I'm referring to this https://github.com/kubernetes/api/blob/v0.21.5/networking/v1/generated.proto#L493-L503

Reply:
addressed 👍